### PR TITLE
[PAGOPA-2104] feat: Add new recovery by IUV API

### PR DESCRIPTION
**Pull request by cap-ang**
#### List of Changes

- A new API (Controller and Service) has been added that given an `IUV` and an `CI` attempts to retrieve the `RPT`.

The function, in detail, checks whether the RPT is interrupted in one of the states that indicate an interrupted RPT. After that, it is checked for the presence of a `SEND_RT_SUCCESS` and if not, sent the `RT` taking as input the `sessionId`

#### Motivation and Context

- The purpose is to automate the recovery process given a `CI` and a `IUV`

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/gov/pagopa/wispconverter/controller/RecoveryController.java [hunk 1]
```diff
@@ -57,4 +57,33 @@
...
+    public ResponseEntity<String> recoverReceiptKOForCreditorInstitutionAndIUV(@PathVariable("creditor_institution") String ci,
+                                                                                                @PathVariable("creditor_institution") String iuv,
+                                                                                                @QueryParam("date_from") String dateFrom,
+                                                                                                @QueryParam("date_to") String dateTo) {
+        try {
+            log.info("Invoking API operation recoverReceiptKOForCreditorInstitution - args: {} {} {} {}", ci, iuv, dateFrom, dateTo);
+
+            boolean recovered = recoveryService.recoverReceiptKO(ci, iuv, dateFrom, dateTo);
+            if(recovered)
+                return ResponseEntity.ok(String.format("RPT with CI %s and IUV %s recovered via API", ci, iuv));
```
[the remark this answers is not in the document]
Comment (cap-ang) [reply src/main/java/it/gov/pagopa/wispconverter/controller/RecoveryController.java:76]: solved in https://github.com/pagopa/pagopa-wisp-converter/pull/106/commits/66e430def27e0eb24568ee431aeaf4eddf149962 with regex validation

File: src/main/java/it/gov/pagopa/wispconverter/service/RecoveryService.java [hunk 2]
```diff
@@ -37,27 +35,49 @@ public class RecoveryService {
...
+    private static final String DATE_FORMAT_DAY = "yyyy-MM-dd";
+
+    @Value("${wisp-converter.cached-requestid-mapping.ttl.minutes:1440}")
+    public Long requestIDMappingTTL;
 
     @Value("${wisp-converter.recovery.receipt-generation.wait-time.minutes:60}")
-    private Long receiptGenerationWaitTime;
+    public Long receiptGenerationWaitTime;
+
+    public boolean recoverReceiptKO(String creditorInstitution, String iuv, String dateFrom, String dateTo) {
```
Comment (jacopocarlini) [inline src/main/java/it/gov/pagopa/wispconverter/service/RecoveryService.java:64]: send the `receiptKO` using sessionId instead

**Review by jacopocarlini — CHANGES_REQUESTED** [review]

File: src/main/java/it/gov/pagopa/wispconverter/service/RecoveryService.java (hunk 2 above)
Comment (cap-ang) [reply src/main/java/it/gov/pagopa/wispconverter/service/RecoveryService.java:64]: the `paaInviaRT-` is invoked using the `sessionId`. What do you mean exactly? The API input is `IUV` so in the function you mentioned we know the `IUV` but not the `sessionId`

**Review by jacopocarlini — APPROVED** [review]
LGTM 💪

**Pull request merged**